### PR TITLE
[niina] Refactoing of COMA_HI::aite_hyouka

### PR DESCRIPTION
**Pull request by peria**
BUG= #125

### CODE REVIEW

File: src/cpu/test_lockit/coma.cc [hunk 1]
```diff
@@ -166,75 +167,61 @@ int COMAI_HI::aite_hyouka(const TLColor ba3[6][kHeight], TLColor tsumo[])
...
+            int keshiko_bb = chousei_syoukyo_2(ba_b, setti_basyo, &chain, dabuchk, &unused_value, &score_bb);
             if (ba_b[2][11] != TLColor::EMPTY)
                 continue;
 
-            for (cc = 1; cc < 5; cc++) {
-                nk1 = NORMAL_TLCOLORS[cc - 1];
-                nk2 = NORMAL_TLCOLORS[cc - 1];
-                for (dd = 0; dd < 6; dd++) {
+            for (int cc = 0; cc < 4; ++cc) {
+                TLColor nk1 = NORMAL_TLCOLORS[cc - 1];
```
Comment (mayah) [inline src/cpu/test_lockit/coma.cc]: array out of index access.
cc - 1 => cc

File: src/cpu/test_lockit/coma.cc [hunk 2]
```diff
@@ -166,75 +167,61 @@ int COMAI_HI::aite_hyouka(const TLColor ba3[6][kHeight], TLColor tsumo[])
...
+                TLColor nk2 = NORMAL_TLCOLORS[cc - 1];
```
Comment (mayah) [inline src/cpu/test_lockit/coma.cc]: ditto.

File: src/cpu/test_lockit/coma.cc [hunk 3]
```diff
@@ -261,124 +248,32 @@ int COMAI_HI::aite_hyouka(const TLColor ba3[6][kHeight], TLColor tsumo[])
                 } // dd
             } // cc
 
-            // aaaaaaaa 111119
-            for (cc = 0; cc < 25; cc++) {
+            // Assume putting 2 same puyos vertically.
+            for (int cc = 0; cc < 24; ++cc) {
                 memcpy(ba, ba_b, sizeof(ba));
```
Comment (mayah) [inline src/cpu/test_lockit/coma.cc]: copyField

File: src/cpu/test_lockit/coma.cc [hunk 4]
```diff
@@ -261,124 +248,32 @@ int COMAI_HI::aite_hyouka(const TLColor ba3[6][kHeight], TLColor tsumo[])
...
-                        continue;
-                }
+                // Skip if the destination is unreachable.
+                if (ba[1][11] != TLColor::EMPTY && x == 0)
+                    continue;
+                if (ba[2][11] != TLColor::EMPTY && x != 2)  // game over
+                    continue;
+                if (ba[3][11] != TLColor::EMPTY && x > 3)
+                    continue;
+                if (ba[4][11] != TLColor::EMPTY && x == 4)
```
Comment (mayah) [inline src/cpu/test_lockit/coma.cc]: x == 5 ?

File: src/cpu/test_lockit/coma.cc (hunk 2 above)
Comment (mayah) [inline src/cpu/test_lockit/coma.cc]: I prefer
````
for (TLColor c : NORMAL_TLCOLORS) {
}
````

File: src/cpu/test_lockit/coma.cc [hunk 5]
```diff
@@ -261,124 +248,32 @@ int COMAI_HI::aite_hyouka(const TLColor ba3[6][kHeight], TLColor tsumo[])
                 } // dd
             } // cc
 
-            // aaaaaaaa 111119
-            for (cc = 0; cc < 25; cc++) {
+            // Assume putting 2 same puyos vertically.
+            for (int cc = 0; cc < 24; ++cc) {
```
Comment (mayah) [inline src/cpu/test_lockit/coma.cc]: I prefer

````
for (int x = 0; x < 6; ++x) {
    for (TLColor c : NORMAL_TLCOLORS) {

    }
}
````

**Comment by peria** [comment]
Thank you for your review, but let me update more and squash them.
This CL was WIP.

File: src/cpu/test_lockit/coma.cc [hunk 6]
```diff
@@ -166,220 +167,99 @@ int COMAI_HI::aite_hyouka(const TLColor ba3[6][kHeight], TLColor tsumo[])
...
+            TLRensaResult result_bb = simulate(ba_b);
             if (ba_b[2][11] != TLColor::EMPTY)
                 continue;
 
-            for (cc = 1; cc < 5; cc++) {
-                nk1 = NORMAL_TLCOLORS[cc - 1];
-                nk2 = NORMAL_TLCOLORS[cc - 1];
-                for (dd = 0; dd < 6; dd++) {
-                    if (tobashi_hantei_a(ba_b, dd, nk1, nk2))
+            for (TLColor nk :NORMAL_TLCOLORS) {
```
Comment (mayah) [inline src/cpu/test_lockit/coma.cc]: nit: space after colon.

**Comment by mayah** [comment]
LGTM. you can merge after resolving nit.

**Pull request merged**